### PR TITLE
[IGA] Membrane and truss embedded edge damping calculation

### PR DESCRIPTION
**Pull request by rickyaristio**
**Description**
This PR modifies how to calculate the Rayleigh damping matrix in a more efficient way for the membrane element and truss embedded edge element. It follows the new function introduced in #8944.

### CODE REVIEW

File: applications/IgaApplication/custom_elements/iga_membrane_element.cpp [hunk 1]
```diff
@@ -193,119 +193,6 @@ namespace Kratos
...
-
-        // definition of problem size
-        const SizeType number_of_nodes = r_geometry.size();
-        const SizeType mat_size = number_of_nodes * 3;
-
-        const auto& r_integration_points = r_geometry.IntegrationPoints();
-
-        const IntegrationMethod integration_method = GetGeometry().GetDefaultIntegrationMethod();
-        const GeometryType::ShapeFunctionsGradientsType& r_shape_functions_gradients = GetGeometry().ShapeFunctionsLocalGradients(integration_method);
-
```
Comment (tteschemacher) [inline applications/IgaApplication/custom_elements/iga_membrane_element.cpp:213]: why do you delete all this?

File: applications/IgaApplication/custom_elements/truss_embedded_edge_element.cpp [hunk 2]
```diff
@@ -165,83 +165,6 @@ namespace Kratos
...
-        const auto& r_integration_points = r_geometry.IntegrationPoints();
-
-        const SizeType r_number_of_integration_points = r_geometry.IntegrationPointsNumber();
-
-        // Prepare memory
-        if (mReferenceBaseVector.size() != r_number_of_integration_points)
-            mReferenceBaseVector.resize(r_number_of_integration_points);
-        
-        //get properties
-        array_1d<double, 3> tangents;
```
Comment (tteschemacher) [inline applications/IgaApplication/custom_elements/truss_embedded_edge_element.cpp:190]: idem

File: applications/IgaApplication/custom_elements/iga_membrane_element.cpp (hunk 1 above)
Comment (rickyaristio) [reply applications/IgaApplication/custom_elements/iga_membrane_element.cpp:213]: In the new CalculateDampingMatrix function, we can obtain the stiffness contribution for the damping by calling the CalculateLeftHandSide directly. 

Therefore, the CalculateInitialStiffnessMatrix, which was actually a copy of the CalculateAll function but neglecting calculation of residual vector, is not needed anymore.

Comment (tteschemacher) [reply applications/IgaApplication/custom_elements/iga_membrane_element.cpp:213]: gotcha. I thought that was for avoiding to store the member variables. but it seems those are pre computed.

**Review by tteschemacher — APPROVED** [review]
cool. give it a go!

File: applications/IgaApplication/custom_elements/iga_membrane_element.cpp (hunk 1 above)
Comment (rickyaristio) [reply applications/IgaApplication/custom_elements/iga_membrane_element.cpp:213]: In the new CalculateDampingMatrix function, we can obtain the stiffness contribution for the damping by calling the CalculateLeftHandSide directly. Thus, the CalculateInitialStiffnessMatrix, which was actually a copy of the CalculateAll function but neglecting calculation of residual vector, is not needed anymore.

Comment (tteschemacher) [reply applications/IgaApplication/custom_elements/iga_membrane_element.cpp:213]: but does that mean that it is not the initial stiffness matrix anymore?

Comment (rickyaristio) [reply applications/IgaApplication/custom_elements/iga_membrane_element.cpp:213]: Yeah, I did a mistake. Thanks for asking this.

By using the new CalculateDampingMatrix, the stiffness matrix will be updated for each non-linear step. It is analogous to the coupling condition where we have conservative and non-conservative integration. 

I need to think more about how to compute the initial stiffness more efficiently, also for the Shell3pElement.

**Pull request merged**